api/src/events.rs

```diff
@@ -17,41 +17,65 @@ use crate::{
-            },
-            Some(Event::ProjectWalletCreated(payload)) => {
+        Services::Treasury(TreasuryEventKey { id, .. }, e) => match e.event {
+            Some(TreasuryEvent::ProjectWalletCreated(payload)) => {
                 process_project_wallet_created_event(db, payload).await
             },
-            Some(Event::MintTransfered(payload)) => {
-                process_mint_transfered_event(db, credits, key, payload).await
-            },
-            Some(Event::MintRetried(payload)) => {
-                process_drop_minted_event(db, credits, key, payload).await
-            },
-            Some(Event::DropRetried(payload)) => {
-                process_drop_created_event(db, credits, key, payload).await
+            Some(TreasuryEvent::PolygonCreateDropTxnSubmitted(PolygonTransactionResult {
+                hash,
+                ..
+            })) => process_drop_created_event(db, credits, id, hash).await,
+            Some(TreasuryEvent::PolygonMintDropSubmitted(PolygonTransactionResult {
+                hash,
+                ..
+            })) => process_drop_minted_event(db, credits, id, hash).await,
+            Some(TreasuryEvent::PolygonTransferAssetSubmitted(PolygonTransactionResult {
+                hash,
+                ..
+            })) => process_mint_transferred_event(db, credits, id, hash).await,
+            Some(TreasuryEvent::PolygonRetryMintDropSubmitted(PolygonTransactionResult {
+                hash,
+                ..
+            })) => process_drop_minted_event(db, credits, id, hash).await,
+            Some(TreasuryEvent::PolygonRetryCreateDropSubmitted(PolygonTransactionResult {
+                hash,
+                ..
+            })) => process_drop_created_event(db, credits, id, hash).await,
+            None | Some(_) => Ok(()),
+        },
+        Services::Solana(SolanaNftEventKey { id, .. }, e) => match e.event {
+            Some(SolanaNftsEvent::CreateDropSubmitted(SolanaCompletedTransaction {
+                signature,
+            })) => process_drop_created_event(db, credits, id, Some(signature)).await,
+            Some(SolanaNftsEvent::MintDropSubmitted(SolanaCompletedTransaction { signature })) => {
+                process_drop_minted_event(db, credits, id, Some(signature)).await
             },
+            Some(SolanaNftsEvent::TransferAssetSubmitted(SolanaCompletedTransaction {
+                signature,
+            })) => process_mint_transferred_event(db, credits, id, Some(signature)).await,
+            Some(SolanaNftsEvent::RetryMintDropSubmitted(SolanaCompletedTransaction {
+                signature,
+            })) => process_drop_minted_event(db, credits, id, Some(signature)).await,
+            Some(SolanaNftsEvent::RetryCreateDropSubmitted(SolanaCompletedTransaction {
+                signature,
+            })) => process_drop_created_event(db, credits, id, Some(signature)).await,
```

Comment:
I removed the optional from the solana signature because even if the transaction fails it still has a signature. 

I'm thinking about doing dedicated <blockchain>_create_drop_failed events.

Reply:
If its an rpc error then there is no signature